drafthorse/pdf.py

```diff
@@ -291,3 +258,80 @@ def _facturx_update_metadata_add_attachment(
         )
     metadata_txt_dict = _prepare_pdf_metadata_txt(pdf_metadata)
     pdf_filestream.add_metadata(metadata_txt_dict)
+
+
+def _extract_xml_info(xml_data):
+    """
+    Extract metadata and profile from XML further added to the PDF
+    :param xml_data: XML data
+    :return: Metadata and profile
+    """
+    INVOICE_TYPE_CODE = "380"
+
+    xml_etree = etree.fromstring(xml_data)
+    namespaces = xml_etree.nsmap
+
+    # get metadata
+    date_xpath = xml_etree.xpath(
+        "//rsm:ExchangedDocument/ram:IssueDateTime/udt:DateTimeString",
+        namespaces=namespaces,
+    )
+    date = date_xpath[0].text
+    date_format = date_xpath[0].attrib and date_xpath[0].attrib.get("format") or "102"
+    format_map = {
+        "102": "%Y%m%d",
+        "203": "%Y%m%d%H%M",
+    }
+    date_dt = datetime.strptime(date, format_map.get(date_format, format_map["102"]))
+    number_xpath = xml_etree.xpath(
+        "//rsm:ExchangedDocument/ram:ID", namespaces=namespaces
+    )
+    number = number_xpath[0].text
+    seller_xpath = xml_etree.xpath(
+        "//ram:ApplicableHeaderTradeAgreement/ram:SellerTradeParty/ram:Name",
+        namespaces=namespaces,
+    )
+    seller = seller_xpath[0].text
+
+    doc_type_xpath = xml_etree.xpath(
+        "//rsm:ExchangedDocument/ram:TypeCode", namespaces=namespaces
+    )
+    doc_type = doc_type_xpath[0].text
+    if doc_type != INVOICE_TYPE_CODE:
+        raise Exception(
+            "Invalid doc type! XML value for TypeCode shall be 380 for an invoice."
+        )
```

Comment:
- This is a breaking change. We don't know if there are users of the library out there who e.g. use 381 or 384. Is it specified that it would be wrong to use 381 or 384? (Where?)
- Please try to use more specific exceptions such as ValueError instead of just Exception